File: projecteuler/go/10/10.go
/*

Summation of primes

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.

*/
package main

import (
	"fmt"
	"github.com/jbarham/primegen.go"
	"math/big"
)

// < ceil
func sumOfPrimesBelow(ceil int64) *big.Int {
	sum := big.NewInt(0)

	gen := primegen.New()
	for {
		// unsafe conversion
		prime := int64(gen.Next())
		if prime >= ceil {
			break
		}
		sum = sum.Add(sum, big.NewInt(prime))
	}

	return sum
}

func main() {
	ceil := int64(2000000)
	sum := sumOfPrimesBelow(ceil)
	fmt.Printf("sum of primes below %v = %s\n", ceil, sum)
}

File: projecteuler/julia/10/10.jl
#=

Summation of primes

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.

=#

using Base.Test

function sumOfPrimesBelow(ceil)
  sum = 0
  for i in 1:ceil-1
    if isprime(i)
	    sum += i
	  end
  end
  return sum
end

@time @test sumOfPrimesBelow(10) == 17
@time @test sumOfPrimesBelow(2000000) == 142913828922